pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SubnetType describes the subnet type.
+type SubnetType string
+
+const (
+	SubnetTypePublic  SubnetType = "public"
+	SubnetTypePrivate SubnetType = "private"
+)
+
+// VPCNetworkConfigurationSpec defines the desired state of VPCNetworkConfiguration
+type VPCNetworkConfigurationSpec struct {
+	// Namespaces to be applied
+	AppliedToNamespaces []string `json:"appliedToNamespaces,omitempty"`
+	// Tier0 gateway or Tier0 VRF path
+	// +kubebuilder:validation:Required
+	Tier0Gateway string `json:"tier0Gateway"`
+	// Edge cluster paths on which the networking elements will be created
+	// +kubebuilder:validation:Required
+	EdgeClusters []string `json:"edgeClusters"`
+	NSXTProject  string   `json:"nsxtProject,omitempty"`
+	// +kubebuilder:validation:MinItems=0
+	// +kubebuilder:validation:MaxItems=5
```

Comment:
How we get "5"?

Reply:
In NSX VPC API definition, https://gitlab.eng.vmware.com/core-build/nsx/-/blob/nsx-keeper/mp/policy/policy-api/src/main/resources/api_spec/root/nsx-policy/policy-vpc.yml,  maxItem for public_ipv4_pool is 5, the same for private_ipv4_pool.  Set to 5 here to keep consistent with NSX VPC API.

public_ipv4_pool:
      type: array
      items: {type: string}
      title: PolicyPath of public ip block
      maxItems: 5
      description: |
        IP block used for allocating CIDR blocks for public subnets. IP block must be defined by the Org admin.
      required: false
      x-vmw-cross-reference:
        - {relationshipType: VPC_PUBLIC_IPPOOL_RELATIONSHIP, leftType: [Vpc], rightType: [IpAddressBlock]}

Reply:
Ok.